--- a/node_modules/@theia/core/lib/browser/messaging/ws-connection-provider.js
+++ b/node_modules/@theia/core/lib/browser/messaging/ws-connection-provider.js
@@ -37,7 +37,7 @@
 inversify_1.decorate(inversify_1.unmanaged(), common_1.JsonRpcProxyFactory, 0);
 exports.HttpFallbackOptions = Symbol('HttpFallbackOptions');
 exports.DEFAULT_HTTP_FALLBACK_OPTIONS = {
-    allowed: true,
+    allowed: false,
     maxAttempts: 2,
     errorTimeout: 5000,
     pollingTimeout: 5000,
